src/dotnet/UI.Blazor/Services/Escapist.cs

```diff
@@ -0,0 +1,17 @@
+namespace ActualChat.UI.Blazor.Services;
+
+public sealed class Escapist
+{
+    private readonly Func<EscapistSubscription> _factory;
+
+    public Escapist(Func<EscapistSubscription> factory)
+        => _factory = factory;
+
+    public async Task<IAsyncDisposable> SubscribeAsync(Action action, CancellationToken token)
+    {
+        if (action == null) throw new ArgumentNullException(nameof(action));
```

Comment:
We rarely do this - i.e. I guess we don't do null checks when the argument is non-nullable + null will still trigger an error later. Sorta "too much of a fool-proof" thing.

Reply:
done